src/main/java/org/cloudburstmc/server/network/BedrockInterface.java

```diff
@@ -101,7 +101,9 @@ public void onUnhandledDatagram(ChannelHandlerContext ctx, DatagramPacket packet
     }
 
     @Override
-    public void setName(String name) {
+    public void setMotd(String motd0, String subMotd0) {
+        String name = motd0 + "!@#" + subMotd0;
```

Comment:
Instead of making concatenation and then splitting it 2 lines later, you should replace code that uses array to use motd and submotd variables.

Reply:
done, concat removed